File: Agent.py
import random
from abc import ABC, abstractmethod
from typing import Tuple

class Agent(ABC):
    
    """_summary_
        @Agent class abtraite
        @param pos : Tuple[int,int]
    """
    def __init__(self, pos:Tuple[int, int]):
        
        self.pos = pos
        self.age = 0
        self.energie = 100
        
    def getPos(self):
        return self.pos
    
    @abstractmethod
    def reproduce(self, other):
        pass
    
    @abstractmethod
    def reproduce_cooldown(self):
        pass

File: Agent.java

public abstract class Agent{

    private Position pos;
    private boolean canReproduce;
    private int energie;
    private int age;

    public Agent(Position pos) {

        this.pos = pos;
        this.canReproduce = false;
        this.age = 0;
        this.energie = 100;
    }

    public Position getPos(){
        return this.pos;
    }

    public boolean isCanReproduce() {
        return canReproduce;
    }

    public int getEnergie() {
        return energie;
    }

    public int getAge() {
        return age;
    }

    public abstract void reproduce();
    public abstract void reproduceCooldaown();
}